pages/data-providers/data-provider.jsx

```diff
@@ -1,7 +1,102 @@
-import React from 'react'
+import React, { useEffect, useContext, useRef } from 'react'
+import { useRouter } from 'next/router'
 
+import { withGlobalStore, GlobalContext } from 'store'
 import DataProviderPageTemplate from 'templates/data-provider'
+import useDebouncedEffect from 'hooks/use-debounced-effect'
 
-const DataProviderPage = () => <DataProviderPageTemplate />
+// TODO: Pages should be just a thin layer of NextJS.
+//       Find a better place for it.
+const useDataProviderController = () => {
+  const formRef = useRef(null)
+  const router = useRouter()
+  const { dataProvider } = useContext(GlobalContext)
 
-export default DataProviderPage
+  useDebouncedEffect(
+    () => {
+      if (dataProvider.query === '') return
+      if (!formRef.current?.dataProviderUrl.checkValidity()) {
+        formRef.current.dataProviderUrl.reportValidity()
+        return
+      }
+      dataProvider.retrieve()
+    },
+    [dataProvider.query],
+    500
+  )
+
+  useEffect(() => {
+    if (router.query?.dataProviderUrl)
+      dataProvider.query = router.query?.dataProviderUrl
+  }, [router.query?.dataProviderUrl])
+
+  return [formRef, dataProvider]
+}
+
+const getHelperMessage = ({ query, created, duplicated, error }) => {
+  if (error) {
+    return {
+      message:
+        'We could not find any OAI-PMH endpoint.' +
+        ' Please try it with another URL.',
+      variant: 'error',
+    }
+  }
+
+  if (created) {
+    return {
+      message: `We found OAI-PMH endpoint at ${created.url}`,
```

Comment:
`entity.url` is the entity's permalink, something like `api.core.ac.uk/collection/{id}`. For the OAI-PMH endpoint URL, we have another property. I am not sure about the name though.

```suggestion
      message: `We found OAI-PMH endpoint at ${created.oaiPmhUrl}`,
```

Reply:
I need to wait for API for these changes

Reply:
I was not sure you did this intentionally or you missed the context. Sorry